### ISSUE TITLE
Try&Catch native iOS exceptions crashes the app with AppCenter Crashes enabled

### ISSUE THREAD

**Issue by busec0** [microsoft/appcenter-sdk-dotnet#1639]
### **Description**

Trying to catch a native iOS exception with AppCenter crashes enabled will lead to a crash. Without enabling AppCenter, the try-catch flow works as expected. 

The snippet below will produce a `NSInvalidArgumentException`

```csharp
public void GenerateNativeCrash()
{
    var dict = new NSMutableDictionary();
    dict.LowlevelSetObject(IntPtr.Zero, IntPtr.Zero);
}
```

And trying to catch this with:

```csharp
public void MyMethod() 
{
    try 
    {
        GenerateNativeCrash(); 
    }
    catch (Exception exception)
    {
        System.Diagnostics.Debug.WriteLine(exception);
    }
}
```

Will actually crash the app when AppCenter Crashes is enabled. Disabling AppCenter, the catch will actually work, the app will not crash and the exception will be printed to the application output. 

Attached a Sample App as well. 

### **Details**

1. What is your app platform (Xamarin.Android or Xamarin.iOS or UWP)?
    - Xamarin.iOS
2. If using Xamarin.Forms or if using portable/shared code to call our SDK APIs, are you using shared project, PCL code or .NET standard code for the application? Which .NET standard version or which PCL profile?
    - .NET standard 2.0.
2. Which SDK version are you using?
    - Microsoft.AppCenter.Crashes 4.5.0
4. Which OS version did you experience the issue on?
    - Tried with iOS 15.3.1 (real device) and iOS 15.4 simulator. 
5. What device version did you see this error on?  Were you using an emulator or a physical device?
    - Both simulator and physical device. 
6. What third party libraries are you using? 
    - Xamarin.Forms, Xamarin.Essentials 
7. Please enable verbose logging for your app using `AppCenter.LogLevel = LogLevel.Verbose` before your call to `AppCenter.Start(...)` and include the logs here:

[applogs.txt](https://github.com/microsoft/appcenter-sdk-dotnet/files/8483096/applogs.txt)
[AppCenterNativeiOSCrashSample.zip](https://github.com/microsoft/appcenter-sdk-dotnet/files/8483097/AppCenterNativeiOSCrashSample.zip)

### System Details 

=== Visual Studio Community 2019 for Mac ===

Version 8.10.22 (build 11)
Installation UUID: ed503438-2289-4f06-bf94-b63070818086
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000174

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.174 (2020-02/adf1bc4335d) (64-bit)
	Package version: 612000174

=== Roslyn (Language Service) ===

3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb

=== NuGet ===

Version: 5.9.0.7134

=== .NET SDK (x64) ===

SDK: /usr/local/share/dotnet/x64/sdk/6.0.201/Sdks
SDK Versions:
	6.0.201
	6.0.102
	5.0.407
	5.0.406
	5.0.405
	3.1.418
	3.1.417
	3.1.416
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/x64/dotnet
Runtime Versions:
	6.0.3
	5.0.16
	5.0.15
	3.1.24
	3.1.23

=== .NET Core 3.1 SDK ===

SDK: 3.1.418

=== .NET 5.0 SDK ===

SDK: 5.0.407

=== Xamarin.Profiler ===

Version: 1.7.0.0
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin.Android ===

Version: 12.2.4.3 (Visual Studio Community)
Commit: xamarin-android/d17-1/fc1125e
Android SDK: /Users/user/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		5.0 (API level 21)
		7.1 (API level 25)
		8.0 (API level 26)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 33.0.1
SDK Build Tools Version: 30.0.3

Build Information: 
Mono: adf1bc4
Java.Interop: xamarin/java.interop/d17-1@7ac33610
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.37.1@0893b3b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d17-1@db125a7

=== Eclipse Temurin JDK ===

Java SDK: Not Found

=== Android SDK Manager ===

Version: 16.10.0.13
Hash: 1b81df5
Branch: remotes/origin/d16-10
Build date: 2021-11-12 00:17:32 UTC

=== Android Device Manager ===

Version: 16.10.0.15
Hash: 89dcc0b
Branch: remotes/origin/d16-10
Build date: 2021-11-12 00:17:52 UTC

=== Xamarin Designer ===

Version: 16.11.0.60
Hash: 56f9b80b0
Branch: remotes/origin/d16-11
Build date: 2021-12-15 01:44:16 UTC

=== Apple Developer Tools ===

Xcode 13.3 (20102)
Build 13E113

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 15.8.0.3 (Visual Studio Community)
Hash: eb7f8ed35
Branch: xcode13.3
Build date: 2022-03-30 03:16:09-0400

=== Build Information ===

Release ID: 810220011
Git revision: 16b49eb779f7def4523bbc8f4407628cf8dbb4b5
Build date: 2022-04-07 05:04:03-04
Build branch: release-8.10

=== Operating System ===

Mac OS X 12.3.0
Darwin 21.4.0 Darwin Kernel Version 21.4.0
    Mon Feb 21 20:35:58 PST 2022
    root:xnu-8020.101.4~2/RELEASE_ARM64_T6000 x86_64

=== Enabled user installed extensions ===

XAML Styler 2.0.1

**Comment by DmitriyKirakosyan** [comment]
Hi @busec0 , thank you for reporting this issue and the detailed repro steps!
I'm marking it as a bug.

**Comment by busec0** [comment]
Hi @DmitriyKirakosyan, any update on this?

**Comment by DmitriyKirakosyan** [comment]
Hi @busec0 , It is still in our queue. I'll update this thread on the progress.

**Comment by busec0** [comment]
Issue is here: https://github.com/microsoft/plcrashreporter/issues/236

**Comment by DmitriyKirakosyan** [comment]
The fix has been released in v4.5.3. Closing.
